bridge/contracts/DepositNotifier.sol

```diff
@@ -0,0 +1,29 @@
+// SPDX-License-Identifier: MIT-open-group
+pragma solidity ^0.8.0;
+
+import "contracts/utils/ImmutableAuth.sol";
+
+/// @custom:salt DepositNotifier
+/// @custom:deploy-type deployUpgradeable
+contract DepositNotifier is ImmutableFactory {
+    uint256 internal _nonce = 0;
+    uint256 internal immutable _networkId;
+
+    event Deposited(
+        uint256 nonce,
+        address ercContract,
+        address owner,
+        uint256 number, // If fungible, this is the amount. If non-fungible, this is the id
+        uint256 networkId
+    );
+
+    constructor(uint256 id) ImmutableFactory(msg.sender) {
+        _networkId = id;
+    }
+
+    function doEmit(bytes32 salt, address ercContract, uint256 number, address owner) onlyFactoryChildren(salt) public {
```

Comment:
access control should only allow bridge pools. I think gustavo has a method of generating salts for his deterministic bridge pool address.